### PR TITLE
Add debugging workshop

### PR DESCRIPTION
**Pull request by Dedekind561**
Adds a debugging workshop with the following:

- 19 broken problems. Each problem should have:
  - `solution.js` with a broken implementation that needs fixing
  -  `spec.md` with a problem statement, functional specification and example function calls for reference

- `readme.md` with an order the problems should be tackled in (increasing in level of difficulty):
  - brief debugging guide
  - warm up
  - core
  - challenge

This workshop also contains a `devcontainer.json` file with node installed at `20.17.1` and eslint at `v9.9.0`

### CODE REVIEW

File: 12-hour-clock/solution.js
Comment (sierrajulietromeo) [inline 12-hour-clock/solution.js:1]: This one would be good for after Launchpad Thursday onwards.

File: body-mass-index/spec.md
Comment (sierrajulietromeo) [inline body-mass-index/spec.md:1]: I'm always cautious about using scenarios like this that refer to the body, the way people look etc.  Even though it's generic, we as of yet don't know the details of the apprentices who will be joining us.

File: find-century/solution.js
Comment (sierrajulietromeo) [inline find-century/solution.js:1]: Nice!

File: body-mass-index/spec.md
Comment (Dedekind561) [reply body-mass-index/spec.md:1]: Yeah I had similar feelings too. I'll swap it out for something else like convert to farenheit or something like that

**Review by sierrajulietromeo — COMMENTED** [review]
Thanks a lot for this Mitch! 👏
Based on the content and quite a bit of string manip etc, I'm going to move this session to be the Monday of week 2.

I toyed around with moving string manip content to earlier in the week, so we could do your session earlier,  but the issue is that currently task 1 comes out on the Wednesday of the first week (which is based on the content of the first two days), as learners want to get stuck in to something that 'matters' and get some points on the board, and feel like they are making progress.      Another issue is that we are remote on Fridays, which brings it's own challenges and I would prefer that your session was done in person.

What we'll do in the first week, is model some informal use of console.logging etc when live coding etc, so the learners get some exposure to it in the first week!

Thanks again!

**Pull request merged**